internal/kubernetes/operator/deployment/deployment_test.go

```diff
@@ -218,10 +219,13 @@ func TestBuildAtlasAdvancedDeployment(t *testing.T) {
 				},
 			},
 			Spec: akov2.AtlasDeploymentSpec{
-				Project: akov2common.ResourceRefNamespaced{
+				Project: &akov2common.ResourceRefNamespaced{
 					Name:      strings.ToLower(projectName),
 					Namespace: targetNamespace,
 				},
+				ExternalProjectRef: &akov2.ExternalProjectReference{
+					ID: projectID,
+				},
```

Comment:
This is an invalid result, these are mutually exclusive properties.

Reply:
Will revisit the approach.